openedx/xblock-drag-and-drop-v2
# fix: Support course-wide "Show Answer" setting

## Description

This PR fixes the global setting for the Show Answer button

The global setting for the Show Answer button is not working for the d&dv2 block; we have it set as "Finished" in the advanced settings (so the button should be clickable after they have used all attempts on the problem or they have correctly answered the problem.), but in d&d v2 blocks, it's greyed out until all attempts have been used.

## Testing instructions

- Login to Studio shell, switch to edxapp virtual env
- Go to /edx/src/xblock-drag-and-drop-v2 and run pip install -e .
- Go to "Settings" -> "Advanced Settings" in studio and the search for "Show Answer" and set it to finished.
- Go to Studio UI and create a new unit
- Select Problem > Advanced > Drag and Drop, to create a v2 drag and drop XBblock
- Click Edit and change mode to Assessment
- Click Continue twice and Save
- Click on View Live Version
- Show Answer button should be clickable after used all attempts on the problem or they have correctly answered the problem

## Supporting information

- [x] Fix the global setting for the Show Answer button
- [x] Fix ```showanswer``` setting failing tests
- [x] Add tests with different "Show answer" statues
